metricbeat/autodiscover/builder/hints/metrics.go

```diff
@@ -202,6 +203,26 @@ func (m *metricHints) getHostsWithPort(hints common.MapStr, port int) ([]string,
 	return result, true
 }
 
+func (m *metricHints) checkHostPort(h string, p int) bool {
+	port := strconv.Itoa(p)
+
+	index := strings.LastIndex(h, ":"+port)
+	// Check if host contains :port. If not then return false
+	if index == -1 {
+		return false
+	}
+
+	// Check if the host ends with :port. Return true if yes
+	end := index + len(port) + 1
+	if end == len(h) {
+		return true
+	}
+
+	// Check if the character immediately after :port. If its not a number then return true.
+	// This is to avoid adding :80 as a valid host for an event that has port=8080
+	return h[end] < '0' || h[end] > '9'
```

Comment:
I guess that this is to cover cases where we accept urls or DSNs (for example a MySQL DSN could be `"tcp(${data.host}:3306)/"`)?
Could you also add a test case with an example where there is a hint with `:port`, that has some non-numeric character afterwards?